claripy/simplifications.py

```diff
@@ -195,6 +197,22 @@ def eq_simplifier(a, b):
         if a.op == 'BVV' and b.op != 'BVV':
             return b == a
 
+        # expr - c == 0    ->   expr == c
+        if a.op == "__sub__" and a.args[1].op == "BVV" and b.op == "BVV":
+            return a.args[0] == a.args[1] + b
+
+        # 1 ^ expr == 0     ->   expr == 1
+        if a.op == '__xor__' and isinstance(b.args[0], int) and b.args[0] == 0:
+            if a.args[0].op == 'If' and a.args[1].args[0] == 1:
+                return a.args[0] == 1
+            elif a.args[1].op == 'If' and a.args[0].args[0] == 1:
+                return a.args[1] == 1
+
+        if a.op == 'Concat' and isinstance(b.args[0], int) and b.args[0] == 0 and \
+                a.args[0].args[0] == 0 and a.args[1].op == '__invert__':
+            return a.args[1].args[0] == 1
```

Comment:
The above 3 lines to me read as transform: `Concat[ op1[0, ...], __invert__[x, ...], ... ] == b.op[0, ...]` into `x == 1`. Where `op1` and `b.op` are unknown ops and where `...` indicates that there may or may not be more arguments, it is not restricted. This seems wrong